source/simulator/assembly.cc

```diff
@@ -1419,7 +1419,8 @@ namespace aspect
                    * on the direction of the flow. Thus the flow-dependent terms below are
                    * only calculated if the edge is an inflow edge.
                    */
-                  const bool inflow = ((current_u * scratch.face_finite_element_values->normal_vector(q)) < 0.);
+                  const bool inflow = (advection_field.is_temperature()
+                                       || ((current_u * scratch.face_finite_element_values->normal_vector(q)) < 0.));
```

Comment:
I would be careful here. Of course in forward calculations the diffusion coefficient is always non-zero, but an often used approach to approximate backward advection is to set the diffusion coefficient to zero (because you can not reverse diffusive processes) and reverse the sign of the gravity. @jaustermann did some tests in that direction during the hackathon, and if it is not much effort then I would prefer keeping the possibility to set up backward advection models with discontinuous elements.